proto/directions.proto

```diff
@@ -0,0 +1,94 @@
+package valhalla;
+
+message Waypoint {
+  optional float lat = 1;
+  optional float lon = 2;
+  optional string type = 3;
+  optional uint32 bearing = 4;
+  optional string name = 5;
+  optional string street = 6;
+  optional string side_of_street = 7;
+}
+
+message BoundingBox {
+  optional float min_lat = 1;
+  optional float min_lon = 2;
+  optional float max_lat = 3;
+  optional float max_lon = 4;
+}
+
+message StreetName {
+  optional string text = 1;  // Text of street name, examples: MD 43 East, White Marsh Boulevard 
+  optional bool is_ref = 2;  // true if street name is a ref, examples: true, false
+}
+
+message Step {
+  optional float distance = 1;
+  optional uint32 duration = 2;
+  optional uint32 geometry_index_begin = 3;      // begin geometry inclusive index
```

Comment:
Whats the purpose in having a start and an end? Could we cut down on a few bytes by only having an end and assuming the start is `end - 1`?

Reply:
The geometry index of a step points to the coordinates in the geojson line array of the geometry of the whole route that represent the step.

A step on a curvy road might span over a couple of indices and not just one line between coordinates - so it is not possible to just get rid of `begin` and `end`. Another design could be `begin` and `length` to know where it stops but that does not save us anything :)

Reply:
I think @mcwhittemore is saying that you assume the start is the last step's `end - 1` but I don't think that works either, because I think some maneuvers are longer than 1 point?